test/test-framework-spec.md

```diff
@@ -303,10 +351,23 @@ Annotation value fields :
 *   Mock object 
 
     ```ballerina
-    http:Client mockClient = <http:Client> test:mock(http:Client, mockObj = new);
+    http:Client mockClient = test:mock(http:Client, new MockHttpClient());
     ```
 
 *   Mock function 
+    
+    Compile-time mocking
```

Comment:
We call this legacy mocking right?

Reply:
But is it correct to call it legacy mocking?
This is the most widely used mocking when testing services and clients right?

Reply:
Used the terms static and dynamic mocking.